### DISCUSSION TITLE
DependencyMatcher and quantifiers

### DISCUSSION THREAD

**Discussion by DavidGerva** [explosion/spaCy#7782] [Help: Coding & Implementations] [answered]
I am working with the new **DependencyMatcher** and I am using a pattern similar to this one:

```
    pattern2 = [
        {
            "RIGHT_ID": "anchor_verb",
            "RIGHT_ATTRS": {"LEMMA": "improved"}
        },
        {
            "LEFT_ID": "anchor_verb",
            "REL_OP": ">",
            "RIGHT_ID": "verb_subject",
            "RIGHT_ATTRS": {"DEP": "nsubj"},
        },
        {
            "LEFT_ID": "anchor_verb",
            "REL_OP": ">",
            "RIGHT_ID": "verb_object",
            "RIGHT_ATTRS": {"DEP": "dobj"},
        },
        {
            "LEFT_ID": "verb_object",
            "REL_OP": ">",
            "RIGHT_ID": "verb_object_modifier",
            "RIGHT_ATTRS": {"DEP": {"IN": ["amod", "compound"]}},
        }
    ]
```

I would like to specify one of the token above as optional and I don't know how to do it. E.g. I want to find matches in the sentence
_"the object improves the mechanical strength of the system"_ and the pattern above does the job.

But I also would like to find matches in this sentence:
_"the object improves the strength of the system"_, in other words I would like to set the last token as optional.
Is it possible to do it? And how?

In the "standard" matcher I would set the key  `"OP": "*"` as explained [here](https://spacy.io/api/matcher) in the "operators and quantifier" section, but in the DependencyMatcher it doesn't work.

**Comment by adrianeboyd** [comment, marked as answer]
Hi, operators for the semgrex relations haven't been implemented yet, so currently the only way to do this would be with two separate patterns.

**Reply by DavidGerva** [reply]
Thank you!
